src/styles/app-select/AppSelect.tsx

```diff
@@ -0,0 +1,204 @@
+import { Icon } from "@iconify/react";
+import { KeyboardEvent, Ref, forwardRef, memo, useRef, useState } from "react";
+import { FixedSizeList, areEqual } from "react-window";
+
+import classNames from "classnames";
+
+import useThrottle from "@/hooks/use-throttle/useThrottle";
+import "@/styles/app-select/AppSelect.scss";
+import { AppSelectProps, ItemProps } from "@/styles/app-select/AppSelect.types";
+import AppSpinner from "@/styles/app-spinner/AppSpinner";
+
+const AppSelect = (
+  {
+    options,
+    onChange,
+    height,
+    width,
+    label,
+    type = "single",
+    renderOptions,
+    overscanCount = 10,
+    itemSize = 50,
+    loadMoreOptions,
+    hasMore,
+    selectedAdornment,
```

Comment:
I suggest following list of available props:

```suggestion
    options,
    onChange,
    label,
    errorMessage,
    helperText,
    multiple,
    renderOption,
    isFetchingOptions,
    controlRef,
    fullWidth,
    ...props
```

Reply:
I'm using width and height in my app-select logic, so I don't see the point of moving it to rest props array. But I'll consider your suggestion with props modifying